src/App/Fossa/Config/SBOM/Analyze.hs

```diff
@@ -0,0 +1,129 @@
+import Options.Applicative.Builder (progDescDoc, strOption)
+import Style (applyFossaStyle, formatStringToDoc, stringToHelpDoc)
+
+data NoUpload = NoUpload
+data JsonOutput = JsonOutput deriving (Generic)
+data SBOMScanDestination
+  = SBOMUploadScan ApiOpts
+  | SBOMOutputStdout
+  deriving (Eq, Ord, Show, Generic)
+
+instance ToJSON SBOMScanDestination where
+  toEncoding = genericToEncoding defaultOptions
+
+data SBOMAnalyzeConfig = SBOMAnalyzeConfig
+  { sbomBaseDir :: BaseDir
+  , sbomApiOpts :: ApiOpts
+  , sbomPath :: SBOMFile
+  , severity :: Severity
+  , sbomRebuild :: DependencyRebuild
+  , sbomTeam :: Maybe Text
+  , sbomRevision :: ProjectRevision
+  }
+  deriving (Eq, Ord, Show, Generic)
+
+instance ToJSON SBOMAnalyzeConfig where
+  toEncoding = genericToEncoding defaultOptions
+
+data SBOMAnalyzeOptions = SBOMAnalyzeOptions
+  { analyzeCommons :: App.Fossa.Config.Common.CommonOpts
+  , team :: Maybe Text
+  , forceRescan :: Flag ForceRescan
+  , sbomFile :: SBOMFile
+  }
+
+instance GetSeverity SBOMAnalyzeOptions where
+  getSeverity SBOMAnalyzeOptions{analyzeCommons = App.Fossa.Config.Common.CommonOpts{optDebug}} = if optDebug then SevDebug else SevInfo
+
+subcommand :: (SBOMAnalyzeOptions -> a) -> Mod CommandFields a
+subcommand f =
+  command
+    "analyze"
+    ( info (f <$> cliParser) $
+        progDescDoc (formatStringToDoc "Scan an SBOM file")
+    )
+
+data ForceRescan = ForceRescan deriving (Generic)
+
+cliParser :: Parser SBOMAnalyzeOptions
+cliParser =
+  SBOMAnalyzeOptions
+    <$> App.Fossa.Config.Common.commonOpts
+    <*> optional (strOption (applyFossaStyle <> long "team" <> short 'T' <> stringToHelpDoc "This SBOM's team inside your organization"))
+    <*> flagOpt ForceRescan (applyFossaStyle <> long "force-rescan" <> stringToHelpDoc "Force sbom file to be rescanned even if the revision has been previously analyzed by FOSSA.")
+    <*> sbomFileArg
+
+mergeOpts ::
+  ( Has Diagnostics sig m
+  , Has ReadFS sig m
+  , Has (Lift IO) sig m
+  ) =>
+  Maybe ConfigFile ->
+  EnvVars ->
+  SBOMAnalyzeOptions ->
+  m SBOMAnalyzeConfig
+mergeOpts cfgfile envvars cliOpts@SBOMAnalyzeOptions{..} = do
+  baseDir <- getCurrentDir
+  let apiOpts = App.Fossa.Config.Common.collectApiOpts cfgfile envvars analyzeCommons
+      severity = getSeverity cliOpts
+      fileLoc = sbomFile
+
+      revOverride =
+        collectRevisionOverride cfgfile $
+          OverrideProject
+            (optProjectName analyzeCommons)
+            (optProjectRevision analyzeCommons)
+            (Nothing)
+
+      forceRescans = if fromFlag ForceRescan forceRescan then DependencyRebuildInvalidateCache else DependencyRebuildReuseCache
+  revision <- getProjectRevision fileLoc revOverride WriteOnly
+  SBOMAnalyzeConfig
+    (BaseDir baseDir)
+    <$> apiOpts
+    <*> pure fileLoc
+    <*> pure severity
+    <*> pure forceRescans
+    <*> pure team
+    <*> pure revision
```

Comment:
It's tedious tow rite, but yes I think this would be better for us to use going forward for larger structures. I think [record punning](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/record_puns.html) is also explicit enough in most cases, but what you suggested here is great.

The real ideal IMO, we'd `newtype` anything that's a more primitive type so that there's a type error if we construct a record wrong but I'm not asking for that here.

Reply:
I agree. It's a bit more work to type, but much more readable. It also avoids bugs if you for some reason reorder the fields in a record.

This would be a good team meeting topic